### ISSUE TITLE
The Autograder may raise undefined method `chomp' for nil:NilClass

### ISSUE THREAD

**Issue by bobbiec** [autolab/Autolab#1131]
As far as I can tell, line 330 will never be triggered.
https://github.com/autolab/Autolab/blob/63a9f8699b6cabeac5812f0156ee60350759154f/app/helpers/assessment_autograde_core.rb#L329-L334

This is because str.lines will never be nil; see example in irb:
```ruby
$ irb
irb(main):001:0> feedback = ''
=> ""
irb(main):002:0> lines = feedback.lines
=> []
irb(main):003:0> autoresult = lines[lines.length-1].chomp
NoMethodError: undefined method `chomp' for nil:NilClass
	from (irb):3
	from /usr/bin/irb:12:in `<main>'
```

This means that on empty script output, the error displayed will be `undefined method 'chomp' for nil:NilClass` rather than `The Autograder returned no output`. This is misleading.

I would suggest changing the condition to throw in line 330 to `if !lines` or `if lines.length == 0`.
```ruby
raise AutogradeError.new("The Autograder returned no output", :autograde_no_output) if !lines 
```

**Comment by victorhuangwq** [comment]
Thank you for figuring this out. We address this issue in a pull request soon.

**Comment by victorhuangwq** [comment]
Addressed in #1146
